java-agent/src/jvmTest/kotlin/com/epam/drill/agent/instrument/reactor/ReactorTest.kt

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright 2020 - 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.agent.instrument.reactor
+
+import com.epam.drill.agent.request.RequestHolder
+import com.epam.drill.common.agent.request.DrillRequest
+import org.junit.Before
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Hooks
+import reactor.core.publisher.Mono
+import reactor.core.publisher.ParallelFlux
+import reactor.core.scheduler.Schedulers
+import reactor.test.StepVerifier
+import java.time.Duration
+import kotlin.test.Test
+
+class ReactorTest {
+
+    @Before
+    fun setUp() {
+        //Reset all installed hooks
+        Hooks.resetOnEachOperator()
+        Schedulers.resetOnScheduleHooks()
+
+        //Init thread local instance
+        RequestHolder.init(false)  //don't use TransmittableThreadLocal
+
+        // InheritableThreadLocal / TransmittableThreadLocal in RequestHolder
+        // enable propagating context to child threads
+        // Warm-up creates child thread in advance, when no drill context is present in thread local yet
+        // later this thread w/o initial context is re-used when calling Schedulers.single()
+        // it won't work in Schedulers.parallel()
+        Mono.just("warm-up")
+            .subscribeOn(Schedulers.single())
+            .block()
+    }
+
+    @Test
+    fun `given Mono class, onAssembly must propagate test session to another threads`() {
```

Comment:
`given Flux class, onAssembly must propagate test session to another threads` - another -> other